pkg/volume/host_path/host_path.go

```diff
@@ -308,3 +323,175 @@ func getVolumeSource(spec *volume.Spec) (*v1.HostPathVolumeSource, bool, error)
 
 	return nil, false, fmt.Errorf("Spec does not reference an HostPath volume type")
 }
+
+func NewHostPathType(pathType string) *v1.HostPathType {
+	hostPathType := new(v1.HostPathType)
+	*hostPathType = v1.HostPathType(pathType)
+	return hostPathType
+}
+
+func newHostPathTypeList(pathType ...string) []*v1.HostPathType {
+	typeList := []*v1.HostPathType{}
+	for _, ele := range pathType {
+		typeList = append(typeList, NewHostPathType(ele))
+	}
+
+	return typeList
+}
+
+type hostPathTypeChecker interface {
+	Exists() bool
+	IsFile() bool
+	MakeFile() error
+	IsDir() bool
+	MakeDir() error
+	IsBlock() bool
+	IsChar() bool
+	IsSocket() bool
+}
+
+func getType(fileInfo os.FileInfo) (v1.HostPathType, error) {
+	mode := fileInfo.Sys().(*syscall.Stat_t).Mode
+	switch mode & syscall.S_IFMT {
+	case syscall.S_IFSOCK:
+		return v1.HostPathSocket, nil
+	case syscall.S_IFBLK:
+		return v1.HostPathBlockDev, nil
+	case syscall.S_IFCHR:
+		return v1.HostPathCharDev, nil
+	}
+
+	return "", fmt.Errorf("only recongnize socket, block device and character device")
+}
+
+// For mock up usage
+var getFileType = getType
+
+type osFileTypeChecker struct {
+	path   string
+	exists bool
+	info   os.FileInfo
+}
+
+func (ftc osFileTypeChecker) Exists() bool {
+	return ftc.exists
+}
+
+func (ftc osFileTypeChecker) IsFile() bool {
+	if !ftc.Exists() {
+		return false
+	}
+
+	return !ftc.info.IsDir()
+}
+
+func (ftc osFileTypeChecker) MakeFile() error {
+	f, err := os.OpenFile(ftc.path, os.O_RDWR|os.O_CREATE|os.O_TRUNC, os.FileMode(0644))
```

Comment:
I dn't think you want TRUNC here - CREATE should suffice, no?

You should check the error for IsExist() and return nil (there's a race and not an error)